File: typescript/src/repositories/book.repository.ts
import { injectable } from 'inversify';
import { BaseRepository } from './base/BaseRepository';
import { Book } from '../entities/Book';
import { ObjectId } from 'mongodb';
import { IRepository } from './interfaces';

export interface IBookRepository extends IRepository<Book> { }

@injectable()
export default class BookRepository extends BaseRepository<Book> implements IBookRepository { }

File: csharp/BookAPI/Repositories/BookRepository.cs
using BookAPI.Entities;
using BookAPI.Repositories.Base;
using BookAPI.Repositories.Interfaces;

namespace BookAPI.Repositories
{
    public class BookRepository: BaseRepository<Book>, IBookRepository
    {
        public BookRepository(AppContext context) : base(context)
        {
        }
    }
}
